src/engine/platform/mixer.cpp

```diff
@@ -47,6 +54,38 @@ struct AudioStruct {
     SDL_AudioDeviceID dev;
 } audio_struct = {};
 
+//TODO(GS) assert delay_time
+Effect create_delay(f32 feedback, f32 delay_time) {
+    auto delay_func = [] (Effect *effect, f32 *buffer, u32 start, u32 len) -> void {
+        for (u32 i = 0; i < len; i++) {
+            //TODO(GS) I am not sure about the math here:       vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+            buffer[(start + i) % TRACK_BUFFER_LENGTH] += buffer[ABS(((s32) start + i - effect->delay.delay_len) % TRACK_BUFFER_LENGTH)] * effect->delay.feedback;
```

Comment:
Apparently % in C/C++ is "remainder" and not "modulo" which means it returns negative values for (-a) % b. Is there a nicer way to do this?

Reply:
`(start + i - effect->delay.delay_len + TRACK_BUFFER_LENGTH) % TRACK_BUFFER_LENGTH`
That way you don't have to do an absolute value which might be a branch instruction depending on how the code is generated.